From 2ea182d9ea197284718b46907ff4075c54710652 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:51 -0400
Subject: [usb] xhci: Add quirks module option

Message-id: <1410887463-20674-220-git-send-email-dzickus@redhat.com>
Patchwork-id: 95165
O-Subject: [RHEL7 PATCH 219/591] xhci: Add quirks module option
Bugzilla: 1069460
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1069460

commit 4e6a1ee72b74ce013f0b31063915a58ba7db2f88
Author: Takashi Iwai <tiwai@suse.de>
Date:   Mon Dec 9 12:42:48 2013 +0100

    xhci: Add quirks module option

    It makes easier for debugging some hardware specific issues.

    Note that this option won't override the value to be set.  That is,
    you can turn quirks on by this option but cannot turn them off if set
    by the driver.

    Signed-off-by: Takashi Iwai <tiwai@suse.de>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/xhci.c b/drivers/usb/host/xhci.c
index d68ec1a..6bc966c 100644
--- a/drivers/usb/host/xhci.c
+++ b/drivers/usb/host/xhci.c
@@ -40,6 +40,10 @@ static int link_quirk;
 module_param(link_quirk, int, S_IRUGO | S_IWUSR);
 MODULE_PARM_DESC(link_quirk, "Don't clear the chain bit on a link TRB");
 
+static unsigned int quirks;
+module_param(quirks, uint, S_IRUGO);
+MODULE_PARM_DESC(quirks, "Bit flags for quirks to be enabled as default");
+
 /* TODO: copied from ehci-hcd.c - can this be refactored? */
 /*
  * xhci_handshake - spin reading hc until handshake completes or fails
@@ -4770,6 +4774,8 @@ int xhci_gen_setup(struct usb_hcd *hcd, xhci_get_quirks_t get_quirks)
  xhci->hcc_params = readl(&xhci->cap_regs->hcc_params);
  xhci_print_registers(xhci);
 
+ xhci->quirks = quirks;
+
  get_quirks(dev, xhci);
 
  /* In xhci controllers which follow xhci 1.0 spec gives a spurious
-- 
1.7.1